[WIP] Day of Defeat: Source entities and other TF2 Branch fixes

SirYodaJedi opened the pull request:
I'm almost done with [my updated Day of Defeat: Source FGD](https://developer.valvesoftware.com/wiki/Day_of_Defeat:_Source_(custom).fgd), so I figured I'd attempt to add the DoD:S entities to this project. I've also bundled some related edits that affect all first-party TF2 Branch games (HL2DM, CSS, TF2, DODS, HLDMS).

This adds a few extra tags: 
* `HLDMS` - Half-Life Deathmatch: Source (has everything HL:S has, plus more, because it runs on TF2's engine version)
* `HL2DM` - Half-Life 2: Deathmatch (surprised this wasn't here already?)
* `PROPLM` - supports lightmapped static props
* `2025MP` - runs on same engine version as TF2 currently does (as of Feb 2025 update) and shares codebase for DLL code. Name suggested by @vrad-exe.
* `MPONLY` - no singleplayer mode; used to hide `newunit` in multiplayer-only games

## Todo
* Editor assets - Done except for comp_team_wall.
* Update code mentions of TF2 such as this one to apply to all 2025MP games: https://github.com/TeamSpen210/HammerAddons/blob/506d542ae5a6a4e99e1ade21040de63850f71701/src/hammeraddons/bsp_transform/globals.py#L68
* Test FGD in Hammer
* Verify `comp_team_wall.py` loads
* Any glaring oversights, as well as Suggestions and recommendations from this repo's maintainers

## Entities
Checkmarked ones are implemented in this PR. Unchecked ones exist in DoD:S, but seem to not be useful for neither Hammer nor the postcompiler.

### Comp entities
* ✔ comp_team_wall - gets renamed to func_team_wall. This allows both `@PointClass` and `@SolidClass` versions to coexist, as func_team_wall is intended to be usable as both.
### Point entities
* ✔ dod_bomb_target
* ✔ dod_control_point
* ✔ dod_location
* ✔ holiday_gift - also added the L4D2 and CS:S versions, which use different models
* ✔ info_player_allies
* ✔ info_player_axis
* ✔ weapon_30cal
* ✔ weapon_amerknife
* ✔ weapon_bar
* ✔ weapon_basebomb - distributed via dod_bomb_dispenser; perhaps could still be useful for point_template
* ✔ weapon_bazooka
* ✔ weapon_c96
* ✔ weapon_colt
* ✔ weapon_frag_ger
* ✔ weapon_frag_ger_live
* ✔ weapon_frag_us
* ✔ weapon_frag_us_live
* ✔ weapon_garand
* ✔ weapon_k98
* ✔ weapon_k98_scoped
* ✔ weapon_m1carbine
* ✔ weapon_mg42
* ✔ weapon_mp40
* ✔ weapon_mp44
* ✔ weapon_p38
* ✔ weapon_pschreck
* ✔ weapon_riflegren_ger
* ✔ weapon_riflegren_ger_live
* ✔ weapon_riflegren_us
* ✔ weapon_riflegren_us_live
* ✔ weapon_smoke_ger
* ✔ weapon_smoke_us
* ✔ weapon_spade
* ✔ weapon_spring
* ✔ weapon_thompson
### Brush entities
* ✔ dod_bomb_dispenser
* ✔ dod_capture_area
* ✔ func_team_wall
* ✔ func_teamblocker
### Logical entities
* ✔ dod_control_point_master
* ✔ dod_scoring
* ✔ info_doddetect
### Code-managed entities and C++ baseclasses
* ✔ base_rocket
* ✔ dod_ammo_box - doesn't seem to work when spawned from point_template; IDK about via VScript
* ✔ dod_bomb_dispenser_icon - spawned at center of dod_bomb_dispenser's bbox; need testing if parenting updates icon location.
* ✔ dod_gamerules
* ❌ dod_objective_resource
* ❌ dod_player_manager
* ❌ dod_ragdoll
* ✔ dod_round_timer - this entity is managed via dod_control_point_master, and can't be controlled directly
* ❌ dod_team_allies
* ❌ dod_team_axis
* ❌ dod_team_manager
* ❌ dod_viewmodel
* ❌ env_grenadetrail - all relevant assets are precached by weapon_basedodgrenade
* ✔ grenade_frag_ger
* ✔ grenade_frag_us
* ✔ grenade_riflegren_ger
* ✔ grenade_riflegren_us
* ✔ grenade_smoke
* ✔ grenade_smoke_ger
* ✔ grenade_smoke_us
* ❌ player - handling via info_player_allies/_axis to avoid complications with other games
* ✔ rocket_bazooka
* ✔ rocket_pschreck
* ✔ weapon_basedodgrenade
* ✔ weapon_dod_base
* ✔ weapon_dod_base_gun
* ✔ weapon_dod_base_melee
* ✔ weapon_dodbaserpg

SirYodaJedi committed:
add all DOD entities except weapons

SirYodaJedi committed:
add holiday_gift

SirYodaJedi committed:
add DoD:S weapons

SirYodaJedi committed:
General TF2 Branch-related updates

SirYodaJedi committed:
fix stray `mincpulevel` choice

TeamSpen210 commented:
Thanks for this! For base classes, you'll want one the  `BaseEntity*` classes - I reorganised them this way to better match how entities are actually defined. Stuff like angles/parenting always exists for instance. For autovisgroups, there's a single config file with all of them, so it's easy to see the big picture. If you've not seen it, I do have [docs](https://github.com/TeamSpen210/HammerAddons/wiki/Unified-FGD) for the syntax changes.

 `@resources` is a custom bit of syntax, it indicates what resources this entity loads itself (other than stuff directly mentioned in the FGD. Might be overkill, but it's useful in some cases. If you don't know leave it out so we know it's still to be done. For `choices`, the extra `engine` tag is used to indicate what the actual data type is once ingame, so we can apply instance transforms, pack resources, etc.

SirYodaJedi commented:
> `@resources` is a custom bit of syntax, it indicates what resources this entity loads itself (other than stuff directly mentioned in the FGD. Might be overkill, but it's useful in some cases.

Is it possible to have resources be conditional based upon the entity's KVs, or do I just list everything? `info_doddetect` has a specific set of soundscripts that it choses from.
![image](https://github.com/user-attachments/assets/861fb37a-0449-4d82-938f-a52064164662)

SirYodaJedi committed:
DoD:S - add kv types for choices and some `@resource`s

SirYodaJedi committed:
forgot german startroundvoice soundscripts in prev commit

SirYodaJedi committed:
DoDS - standardize baseclasses... mostly

still need to merge DoDSWeapon with Weapon

TeamSpen210 commented:
It can be conditional yes, but it's a bit more work I'll probably need to do. You define `func somefunc`, then those are implemented as Python code in the `srctools` library. Probably easier to list everything, main place that's useful is if stuff is built up out of multiple keyvalues or something. Also, for the unchecked ents, internal ones go in `engine`, and can be referenced by an `entity some_ent` resource to indicate it's getting spawned. Up to you if you want to go to that detail though. It's obviously not necessary for the FGDs, but provides information for the postcompiler and other tools.

SirYodaJedi committed:
fix detect_allies_startroundvoice[engine]

SirYodaJedi committed:
forgot holiday_gift

SirYodaJedi committed:
and `func_team*`; this should be all the non-weapon bases now

SirYodaJedi committed:
Add source files for dod_location iconsprite

* based upon Twemoji U+1F4CD; added to CREDITS.md as per attribution requirements
https://github.com/twitter/twemoji
* font from Mapbase
https://github.com/mapbase-source/source-sdk-2013/wiki/Media-Resources

SirYodaJedi committed:
modelsrc for info_player_axis and info_player_allies editor models

SirYodaJedi committed:
add editor models for axis/allies spawn ents

ommited `.sw.vtx` because directx 7 is no longer supported

SirYodaJedi committed:
add `dod_location` iconsprite

vrad-exe commented:
You should really rename `TF2BRANCH` to `2013MP`, since that's really what branch it is internally, the VDC is kind of misleading listing it as separate

SirYodaJedi committed:
add remaining iconsprites

SirYodaJedi commented:
> You should really rename `TF2BRANCH` to `2013MP`, since that's really what branch it is internally, the VDC is kind of misleading listing it as separate

It's a continuation, but incompatible. It's not an actual "branch", no, but it is a separate engine version.

SirYodaJedi committed:
change `TF2BRANCH` to `TF2ENG`
since it's not really a branch. Not calling 2013MP b/c it's very different from 2021's 2013MP, which is still used by a number of mods.

SirYodaJedi committed:
deprecate and remove DoDSWeapon baseclass

vrad-exe commented:
Perhaps `2025MP` would be better then if you want to keep it separate, I don't feel like it really makes sense to refer to it with TF2 though since it's used for a bunch of other games too

SirYodaJedi committed:
add weapon_basebomb

SirYodaJedi commented:
That works; keeps it future-proof in case there's another major update.

SirYodaJedi committed:
change `TF2ENG` to `2025MP`
name suggested by @vrad-exe

SirYodaJedi committed:
add dods to build scripts

SirYodaJedi committed:
missed weapon_colt when removing DoDSWeapon

SirYodaJedi committed:
and the frag grenades
*now* that should be all the baseclasses; I checked all of them this time.

SirYodaJedi committed:
prop_static - po2 choices for lightmap res
Also extended description for generatelightmaps

SirYodaJedi committed:
improve some dod ent descriptions, and don't autopack DCP's `model` KV

SirYodaJedi committed:
add currently implemented ents to visgroups.cfg

SirYodaJedi committed:
add dod_scoring iconsprite, and fix my other icon paths

SirYodaJedi committed:
helps to actually set the iconsprite()...

TeamSpen210 commented:
Yeah `aurovis()` isn't, I need to remove that. Originally I had normal `@AutoVisgroup` blocks, then made the `autovis()` shorthand, then realised the amount of visgroups were inconsistent and excessive. So I had it dump out the full list into the config file we have, deactivated `autovis()` and adjusted the config.

TeamSpen210 commented:
Oh also noticed you mentioned Mapbase's ent sprite font - I do have another option, a script in the repo root that assembles those.

SirYodaJedi commented:
Ah, `spritefont.py`; I see that now. That simplifies things; wish I noticed it sooner.

SirYodaJedi committed:
holiday_gift - inherit from Item

SirYodaJedi committed:
coldworld can be useful for vscript, apparently

SirYodaJedi committed:
func_team_wall switches to SOLID_OBB if angles are defined

SirYodaJedi committed:
add `comp_team_wall` as alias for func_team_wall.

SirYodaJedi committed:
Add `comp_team_wall` to autovisgroup

SirYodaJedi committed:
Idk why keep on calling it team_clip

SirYodaJedi committed:
add most relevant DODS code-managed entities except grenades

SirYodaJedi committed:
Have `info_player_axis`/`_allies` handle `dod_ammo_box` spawning
Technically, `player` handles it, but that's an unnecessary can of worms

SirYodaJedi committed:
fix missing quote mark in worldspawn.fgd

SirYodaJedi committed:
Add code-managed grenades and rockets for postcompiler

SirYodaJedi committed:
fix base_rocket classname

I need to figure out how to run the test scripts locally

SirYodaJedi committed:
Add`CDODPlayer` assets

SirYodaJedi committed:
add `@resources` to a weapons
only weaponscripts, baseclasses, and spawned ents rn; need to go through and check what else is loaded.

SirYodaJedi committed:
fix dupe rocket_bazooka

SirYodaJedi committed:
TF2 -> 2025MP for `trigger_teleport_relative`, `skybox_swapper`, and `_point_worldtext_tf2`

SirYodaJedi committed:
remaining precached assets

SirYodaJedi committed:
fix holiday_gift baseclass

SirYodaJedi committed:
comp_team_wall - obb() -> wirebox()
fixes vanilla hammer compatability

SirYodaJedi committed:
copy skybox_swapper TODO comment to worldspawn + p1 default sky

SirYodaJedi committed:
add `report` to dod_control_point and dod_location

SirYodaJedi commented:
Okay, I tried to run the build scripts locally, but command prompt crashes.

TeamSpen210 commented:
What error do you get? What version of Python are you using?

SirYodaJedi commented:
> What error do you get?

Okay, I had to open command prompt from within command prompt to get it to not "crash", because the `exit` commands from within the `build.bat` file would close the command prompt instance (which I perceived as a "crash"). I see now that I need to install srctools.

> What version of Python are you using?

I recently changed the version on my %path% to 3.13.3.

SirYodaJedi committed:
fix invalid `bbox()` parameters

SirYodaJedi committed:
Merge branch 'dev' into dods_ents

SirYodaJedi commented:
note to self to add the DetonateTime input to DoD:S grenades if I ever get back to this
